src/plugins/data/common/osd_field_types/osd_field_types.ts

```diff
@@ -84,3 +84,43 @@ export const setOsdFieldOverrides = (newOverrides: { [key: string]: any } | unde
 export const getOsdFieldOverrides = (): { [key: string]: any } => {
   return osdFieldOverrides;
 };
+
+/**
+ *  Mapping function from SQL_TYPES to OSD_FIELD_TYPES
+ *
+ *  @param {OPENSEARCH_SQL_TYPES} sqlType
+ *  @return {OSD_FIELD_TYPES}
+ */
+export function castSQLTypeToOSDFieldType(sqlType: OPENSEARCH_SQL_TYPES): OSD_FIELD_TYPES {
+  switch (sqlType) {
+    case OPENSEARCH_SQL_TYPES.BOOLEAN:
```

Comment:
Nice! i think we should add a section about this information in the reference area: https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/query_enhancements/public/query_editor/ppl_docs for sql or update these docs here: https://opensearch.org/docs/latest/search-plugins/sql/datatypes/

since there are functionality implications. thank you so much for handling this!

Reply:
I'm talking to SQL maintainers on this. The documentation from SQL is not updated when it comes to return types, there are some discrepancies:  https://github.com/opensearch-project/sql/issues/1296